pkg/database/database.go

```diff
@@ -149,33 +149,39 @@ func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) err
 	// opencenusus stats.
 	var rawSQL *sql.DB
 	if err := withRetries(ctx, func(ctx context.Context) error {
-		var err error
-		rawSQL, err = sql.Open("ocsql", c.ConnectionString())
+		s, err := sql.Open("ocsql", c.ConnectionString())
 		if err != nil {
+			return fmt.Errorf("failed to open sql connection: %w", err)
+		}
+
+		if err := s.Ping(); err != nil {
```

Comment:
does `s` need to be closed if ping fails?

Reply:
Hmm yea, good point. I wonder if that's why we're exhausting... I refactored this a bit.